gcovr/workers.py

```diff
@@ -56,9 +56,14 @@ def locked_directory(dir_):
     """
     Context for doing something in a locked directory
     """
-    locked_directory.global_object.run_in(dir_)
-    yield
-    locked_directory.global_object.done(dir_)
+    try:
+        locked_directory.global_object.run_in(dir_)
+        yield
+        locked_directory.global_object.done(dir_)
+    except Exception:
+        # unlock the directory and reraise the exception.
+        locked_directory.global_object.done(dir_)
+        raise
```

Comment:
Wow, great find, thank you for fixing this!

FYI the resource management pattern [suggested by the `@contextmanager` docs](https://docs.python.org/3/library/contextlib.html#contextlib.contextmanager) would be:

```python
def locked_directory(dir_):
  locked_directory.global_object.run_in(dir_)
  try:
    yield
  finally:
    locked_directory.global_object.done(dir_)
```

Reply:
I wasn't sure habout the finally here. I'll change this in a new PR.